src/Magnum/Shaders/Test/FlatGLTest.cpp

```diff
@@ -853,6 +860,37 @@ template<UnsignedInt dimensions> void FlatGLTest::construct() {
     MAGNUM_VERIFY_NO_GL_ERROR();
 }
 
+template<UnsignedInt dimensions> void FlatGLTest::constructAsync() {
+    setTestCaseTemplateName(Utility::format("{}", dimensions));
+
+    auto&& data = ConstructData[testCaseInstanceId()];
+    setTestCaseDescription(data.name);
```

Comment:
No need to test all combinations here again (apologies if I haven't been clear on this). Just pick one case from `ConstructData` that has sufficiently non-trivial `flags` (but ideally doesn't need to check for any extensions, to reduce the checks) and replicate it in this test case.

Same for the uniform buffer variant below, there just pick something with non-trivial `materialCount` / `drawCount`.